src/credentials/FabricTable.cpp

```diff
@@ -317,15 +317,25 @@ CHIP_ERROR FabricInfo::VerifyCredentials(const ByteSpan & noc, const ByteSpan &
     NodeId nodeId;
     ReturnErrorOnFailure(ExtractNodeIdFabricIdFromOpCert(certificates.GetLastCert()[0], &nodeId, &fabricId));
 
+    FabricId icacFabric = kUndefinedFabricId;
     if (!icac.empty())
     {
-        FabricId icacFabric = kUndefinedFabricId;
         if (ExtractFabricIdFromCert(certificates.GetCertSet()[1], &icacFabric) == CHIP_NO_ERROR && icacFabric != kUndefinedFabricId)
         {
             ReturnErrorCodeIf(icacFabric != fabricId, CHIP_ERROR_FABRIC_MISMATCH_ON_ICA);
         }
     }
 
+    FabricId rootFabric = kUndefinedFabricId;
+    if (ExtractFabricIdFromCert(certificates.GetCertSet()[0], &rootFabric) == CHIP_NO_ERROR && rootFabric != kUndefinedFabricId)
+    {
+        if (!icac.empty())
+        {
+            ReturnErrorCodeIf(rootFabric != icacFabric, CHIP_ERROR_FABRIC_MISMATCH_ON_ICA);
```

Comment:
Why do we need to do this check?  We already checked that icacFabric == fabricId (at least if icacFabric was not undefined, which we are not allowing here...).  If we now check that rootFabric == fabricId, that will also check that it eqals icacFabric, no?

Reply:
FabricId attribute is optional in both ICAC and RCAC. But if it is present in RCAC then we should verify that if was also present in ICAC. No make it clear that the code if verifying presents of FabricId attribute in ICAC I changed it to the following check: `ReturnErrorCodeIf(icacFabricId == kUndefinedFabricId, ...` and added comment.